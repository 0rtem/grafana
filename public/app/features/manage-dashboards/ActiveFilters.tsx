import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { DashboardQuery } from 'app/types';
import { removeStarredFilter, removeTag, clearFilters } from './state/actions';
import { getDashboardQuery } from './state/selectors';

export interface Props {
  query: DashboardQuery;
  removeStarredFilter: typeof removeStarredFilter;
  removeTag: typeof removeTag;
  clearFilters: typeof clearFilters;
}

export class ActiveFilters extends PureComponent<Props, any> {
  removeTag = tagName => {
    this.props.removeTag(tagName);
  };

  removeStarred = () => {
    this.props.removeStarredFilter();
  };

  clearFilters = () => {
    this.props.clearFilters();
  };

  render() {
    const { query } = this.props;

    return (
      <div className="page-action-bar page-action-bar--narrow">
        <div className="gf-form-inline">
          {query.tag.length > 0 && (
            <div className="gf-form">
              <label className="gf-form-label width-4">Tags</label>
              <div className="gf-form-input gf-form-input--plaintext">
                {query.tag.map((tagName, index) => {
                  return (
                    <span key={`${tagName}-${index}`}>
                      <div
                        onClick={() => {
                          this.removeTag(tagName);
                        }}
                        tag-color-from-name="tagName"
                        className="tag label label-tag"
                      >
                        <i className="fa fa-remove" />&nbsp;{tagName}
                      </div>
                    </span>
                  );
                })}
              </div>
            </div>
          )}
          {query.starred && (
            <div className="gf-form">
              <label className="gf-form-label">
                <div className="pointer" onClick={this.removeStarred}>
                  <i className="fa fa-fw fa-check" /> Starred
                </div>
              </label>
            </div>
          )}

          <div className="gf-form">
            <label className="gf-form-label">
              <div
                className="pointer"
                onClick={this.clearFilters}
                bs-tooltip="'Clear current search query and filters'"
              >
                <i className="fa fa-remove" />&nbsp;Clear
              </div>
            </label>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    query: getDashboardQuery(state.manageDashboards),
  };
}

const mapDispatchToProps = {
  removeStarredFilter,
  removeTag,
  clearFilters,
};

export default connect(mapStateToProps, mapDispatchToProps)(ActiveFilters);
