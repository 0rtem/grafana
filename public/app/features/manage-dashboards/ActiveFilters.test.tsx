import React from 'react';
import { shallow } from 'enzyme';
import { ActiveFilters, Props } from './ActiveFilters';

const defaultQuery = {
  tag: [],
  starred: false,
  mode: 'tree',
  skipStarred: false,
  skipRecent: false,
  folderIds: [],
  query: '',
};

const setup = (propOverrides?: object) => {
  const props: Props = {
    query: defaultQuery,
    toggleFilterOnStarred: jest.fn(),
    removeTagFilter: jest.fn(),
    clearFilters: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<ActiveFilters {...props} />);
  const instance = wrapper.instance() as ActiveFilters;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render tags', () => {
    const { wrapper } = setup({ query: { ...defaultQuery, tag: ['blue', 'green', 'red'] } });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render starred', () => {
    const { wrapper } = setup({ query: { ...defaultQuery, starred: true } });

    expect(wrapper).toMatchSnapshot();
  });
});
