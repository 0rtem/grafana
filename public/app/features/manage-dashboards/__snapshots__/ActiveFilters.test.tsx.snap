// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="page-action-bar page-action-bar--narrow"
>
  <div
    className="gf-form-inline"
  >
    <div
      className="gf-form"
    >
      <label
        className="gf-form-label"
      >
        <div
          bs-tooltip="'Clear current search query and filters'"
          className="pointer"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
           Clear
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`Render should render starred 1`] = `
<div
  className="page-action-bar page-action-bar--narrow"
>
  <div
    className="gf-form-inline"
  >
    <div
      className="gf-form"
    >
      <label
        className="gf-form-label"
      >
        <div
          className="pointer"
          onClick={[Function]}
        >
          <i
            className="fa fa-fw fa-check"
          />
           Starred
        </div>
      </label>
    </div>
    <div
      className="gf-form"
    >
      <label
        className="gf-form-label"
      >
        <div
          bs-tooltip="'Clear current search query and filters'"
          className="pointer"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
           Clear
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`Render should render tags 1`] = `
<div
  className="page-action-bar page-action-bar--narrow"
>
  <div
    className="gf-form-inline"
  >
    <div
      className="gf-form"
    >
      <label
        className="gf-form-label width-4"
      >
        Tags
      </label>
      <div
        className="gf-form-input gf-form-input--plaintext"
      >
        <span
          key="blue-0"
        >
          <div
            className="tag label label-tag"
            onClick={[Function]}
            tag-color-from-name="tagName"
          >
            <i
              className="fa fa-remove"
            />
             
            blue
          </div>
        </span>
        <span
          key="green-1"
        >
          <div
            className="tag label label-tag"
            onClick={[Function]}
            tag-color-from-name="tagName"
          >
            <i
              className="fa fa-remove"
            />
             
            green
          </div>
        </span>
        <span
          key="red-2"
        >
          <div
            className="tag label label-tag"
            onClick={[Function]}
            tag-color-from-name="tagName"
          >
            <i
              className="fa fa-remove"
            />
             
            red
          </div>
        </span>
      </div>
    </div>
    <div
      className="gf-form"
    >
      <label
        className="gf-form-label"
      >
        <div
          bs-tooltip="'Clear current search query and filters'"
          className="pointer"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
           Clear
        </div>
      </label>
    </div>
  </div>
</div>
`;
